dm/worker/config.go

```diff
@@ -266,26 +312,43 @@ func (c *Config) adjustFlavor() error {
 			return terror.ErrNotSupportedFlavor.Generate(c.Flavor)
 		}
 	}
-	// decrypt password
-	clone, err := c.DecryptPassword()
-	if err != nil {
-		return err
+
+	c.Flavor, err = utils.GetFlavor(ctx, db)
+	if ctx.Err() != nil {
+		err = terror.Annotatef(err, "time cost to get flavor info exceeds %s", dbGetTimeout)
+	}
+	return terror.WithScope(err, terror.ScopeUpstream)
+}
+
+func (c *Config) adjustServerID(ctx context.Context, db *sql.DB) error {
+	if c.ServerID != 0 {
+		if c.ServerID < 0 || c.ServerID > maxServerID {
+			return terror.ErrInvalidServerID.Generate(c.ServerID)
+		}
+
+		return nil
+	}
+
+	serverIDs, err := getSlaveServerIDFunc(ctx, db)
```

Comment:
Should we also check with master's server-id too?

Reply:
yes, update in 394d4c2